src/components/grid-layer/grid-layer-component.jsx

```diff
@@ -0,0 +1,104 @@
+import { loadModules } from '@esri/react-arcgis';
+import { useState, useEffect, useRef } from 'react';
+import { useWatchUtils } from 'hooks/esri';
+import { BIODIVERSITY_FACETS_LAYER } from 'constants/biodiversity';
+
+import { createGridCellGraphic, createGraphicLayer, calculateAgregatedGridCellGeometry, cellsEquality } from 'utils/grid-layer-utils';
+
+const GridLayer = ({map, view, setGridCellData, setGridCellGeometry}) => {
+
+  let watchHandle;
+  let queryHandle;
+
+  const watchUtils = useWatchUtils();
+  const [viewExtent, setViewExtent] = useState();
+  const [gridViewLayer, setGridViewLayer] = useState(null);
+  const [gridCellGraphic, setGridCellGraphic] = useState(null);
+  // References for cleaning up graphics
+  const gridCellRef = useRef();
+
+  //Create the graphics layer on mount
+  useEffect(() => {
+    loadModules(
+      [
+        "esri/Graphic",
```

Comment:
I was thinking, maybe we can do something with these strings here, when we do `loadModules()`, e.g. move them to separate file as constants, and then import them in our components? It's more like an open question, ofc it's not an issue for this PR
```
export const GRAPHICS_LAYER = "esri/layers/GraphicsLayer";
```